internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+	state.PersistentGrantIdleTimeout = types.Int64PointerValue(r.PersistentGrantIdleTimeout)
+	state.PersistentGrantIdleTimeoutTimeUnit = types.StringPointerValue(r.PersistentGrantIdleTimeoutTimeUnit)
+	state.PersistentGrantReuseType = types.StringPointerValue(r.PersistentGrantReuseType)
+	state.PersistentGrantReuseGrantTypes = internaltypes.GetStringSet(r.PersistentGrantReuseGrantTypes)
+	state.AllowAuthenticationApiInit = types.BoolPointerValue(r.AllowAuthenticationApiInit)
+	state.BypassApprovalPage = types.BoolPointerValue(r.BypassApprovalPage)
+	state.RestrictScopes = types.BoolPointerValue(r.RestrictScopes)
+	restrictedScopesToSet, respDiags := types.SetValueFrom(ctx, types.StringType, r.RestrictedScopes)
+	diags.Append(respDiags...)
+	state.RestrictedScopes = restrictedScopesToSet
+	state.ExclusiveScopes = internaltypes.GetStringSet(r.ExclusiveScopes)
+	state.AuthorizationDetailTypes = internaltypes.GetStringSet(r.AuthorizationDetailTypes)
+	state.RestrictedResponseTypes = internaltypes.GetStringSet(r.RestrictedResponseTypes)
+	state.RequirePushedAuthorizationRequests = types.BoolPointerValue(r.RequirePushedAuthorizationRequests)
+	state.RequireJwtSecuredAuthorizationResponseMode = types.BoolPointerValue(r.RequireJwtSecuredAuthorizationResponseMode)
+	state.RequireSignedRequests = types.BoolPointerValue(r.RequireSignedRequests)
+	state.RequestObjectSigningAlgorithm = types.StringPointerValue(r.RequestObjectSigningAlgorithm)
+
+	// state.OidcPolicy
+	oidcPolicyToState, respDiags := types.ObjectValueFrom(ctx, oidcPolicyAttrType, r.OidcPolicy)
+	diags.Append(respDiags...)
+	state.OidcPolicy = oidcPolicyToState
+
+	// state.ClientAuth
+	var clientAuthToState types.Object
+	if r.ClientAuth != nil {
+		clientAuthFromPlan := plan.ClientAuth.Attributes()
+		var secretToState basetypes.StringValue
+		secretVal := clientAuthFromPlan["secret"]
+		if secretVal != nil {
+			secretToState = types.StringValue(secretVal.(types.String).ValueString())
+		} else {
+			secretToState = types.StringValue("")
+		}
+
+		var secondarySecretsSetSlice []attr.Value
+		var secondarySecretsObjToState types.Set
+		secondarySecretsFromPlan := clientAuthFromPlan["secondary_secrets"]
+		if secondarySecretsFromPlan != nil && len(secondarySecretsFromPlan.(types.Set).Elements()) > 0 {
+			for _, secondarySecretsFromPlan := range clientAuthFromPlan["secondary_secrets"].(types.Set).Elements() {
+				secondarySecretsAttrVal, respDiags := types.ObjectValueFrom(ctx, secondarySecretsAttrType, secondarySecretsFromPlan)
+				diags.Append(respDiags...)
+				secondarySecretsSetSlice = append(secondarySecretsSetSlice, secondarySecretsAttrVal)
+			}
+			secondarySecretsObjToState, respDiags = types.SetValue(types.ObjectType{AttrTypes: secondarySecretsAttrType}, secondarySecretsSetSlice)
+			diags.Append(respDiags...)
+		} else {
+			secondarySecretsObjToState = types.SetNull(types.ObjectType{AttrTypes: secondarySecretsAttrType})
```

Comment:
Should this be an empty set rather than a null set?

Reply:
I'll change this to an empty set.

Reply:
done

Reply:
Looks like this is still using SetNull

Reply:
whoops! will get this added.